src/pytest_workflow/plugin.py

```diff
@@ -36,35 +39,46 @@ def pytest_collect_file(path, parent):
 class YamlFile(pytest.File):
     """This class collects YAML files and turns them into test items."""
 
+    def __init__(self, path, parent):
+        super(YamlFile, self).__init__(path, parent=parent)
+
     def collect(self):
         with self.fspath.open() as yaml_file:
-            yaml_content = yaml.load(yaml_file)
-        validate_schema(yaml_content)
-        yield WorkflowItem(self.fspath.basename, self, yaml_content)
+            yaml_content = yaml.safe_load(yaml_file)
+        yield WorkflowTestsCollector(self.fspath.basename, self, yaml_content)
 
 
-class WorkflowItem(pytest.Item):
+class WorkflowTestsCollector(pytest.Collector):
     """This class defines a pytest item. That has methods for running tests."""
 
     def __init__(self, name, parent, yaml_content: dict):
+        validate_schema(yaml_content)
         self.yaml_content = yaml_content
+        self.name = name
+        super(WorkflowTestsCollector, self).__init__(name, parent=parent)
 
-        super(WorkflowItem, self).__init__(name, parent)
-
-    def runtest(self):
-        """Run test runs the item test
-        We use the workflow_run fixture here to run the workflow"""
+    def collect(self):
+        """Run the workflow and start the tests"""
+        # Make sure workflow is run in a temporary directory
+        tempdir = tempfile.mkdtemp(prefix="pytest_wf")
+        copy_tree(os.getcwd(), tempdir)
         workflow = Workflow(
             executable=self.yaml_content.get("executable"),
-            arguments=self.yaml_content.get("arguments"))
+            arguments=self.yaml_content.get("arguments"),
+            cwd=tempdir)
         workflow.run()
-
-        # Here all the assertions are done. This is butt-ugly. Preferably
-        # Some stuff is parameterized or something.
-        assert workflow.exit_code == 0  # We may want to allow for failing workflows later and make this configurable in the yaml.
-        for test_file in self.yaml_content.get("results", {}).get("files", []):
-            assert Path(test_file.get("path")).exists()
-
+        workflow_tests = [
+            WorkflowFilesTestCollector(
+                self.name, self,
+                self.yaml_content.get("results", {}).get("files", []),
+                tempdir)]
+        for test in workflow_tests:
+            yield test
+        # TODO: Figure out proper cleanup.
+        # If tempdir is removed here, all tests will fail.
```

Comment:
Perhaps make an issue for this, so we can refer to it in the future. I have some ideas as to how to fix this, but that's probably out of scope for this PR.

Reply:
:+1: